va/validation-authority.go

```diff
@@ -503,25 +504,31 @@ func (va *ValidationAuthorityImpl) validate(authz core.Authorization, challengeI
 	va.RA.OnValidationUpdate(authz)
 }
 
-func (va *ValidationAuthorityImpl) validateChallengeAndCAA(identifier core.AcmeIdentifier, challenge core.Challenge, regID int64) ([]core.ValidationRecord, *probs.ProblemDetails) {
+func (va *ValidationAuthorityImpl) validateChallengeAndCAA(ctx context.Context, identifier core.AcmeIdentifier, challenge core.Challenge, regID int64) ([]core.ValidationRecord, *probs.ProblemDetails) {
 	ch := make(chan *probs.ProblemDetails, 1)
 	go func() {
-		ch <- va.checkCAA(identifier, regID)
+		ch <- va.checkCAA(ctx, identifier, regID)
 	}()
 
-	validationRecords, err := va.validateChallenge(identifier, challenge)
+	// TODO(#1292): send into another goroutine
+	validationRecords, err := va.validateChallenge(ctx, identifier, challenge)
 	if err != nil {
 		return validationRecords, err
 	}
 
-	caaProblem := <-ch
-	if caaProblem != nil {
-		return validationRecords, caaProblem
+	select {
+	case <-ctx.Done():
+		// TODO(#1292): stat and log
+		return validationRecords, probs.ServerInternal("Unable to receive validation data in time")
```

Comment:
Does this turn CAA timeouts into ServerInternal errors? I think not, so long as the DNS timeout is lower than the timeout on the context passed in. But it's worth commenting about that. We definitely want to return a more informative message for CAA timeouts, which are fairly common.